### ISSUE TITLE
\' shoud be replaced by separator

### ISSUE THREAD

**Issue by sfermigier** [un33k/python-slugify#17]
In your example:

~~~python
txt = 'C\'est déjà l\'été.'
r = slugify(txt)
self.assertEqual(r, "cest-deja-lete")
~~~

I believe the result should be 'c-est-deja-l-ete', not 'cest-deja-lete'.

**Comment by un33k** [comment]
Let me have a look at a proper solution that works for all languages.

**Comment by fabiocaccamo** [comment]
+1 
The actual behavior is wrong, in many languages the single quote is used to compact 2 different words.

To fix this just replace [this](https://github.com/un33k/python-slugify/blob/master/slugify/slugify.py#L122):
```
text = QUOTE_PATTERN.sub('', text.lower())
```
with this:
```
text = QUOTE_PATTERN.sub('-', text.lower())
```

**Comment by un33k** [comment]
Now, the reason the single quote was replaced in the first place was for the following reason.

Reason for removal of ( ' ):
Computer (English) -> компьютер (Russian) -> komp'yuter (converted)
There is no ( ' ) in the Russian version, so we can't remove it prior to our process.
Once the process is complete, then we scan to see if there are any unwanted characters.
At this point we have (komp'yuter), which cannot be written as (komp-yuter) just like compu-ter is meaningless in English. (this was brought to my attention as I don't speak Russian)

One solution would be to split the ( ' ) lookup into two separate processes.
Preprocess: replace ( ' ) with ( - )
Postprocess: remove ( ' )

But even that might not work with the example above as they ( ' )s are escaped. We need to look for the possibility of also pre-process unsnapping, the manual way!

**Comment by fabiocaccamo** [comment]
My solution idea is exactly like the one you just suggested:

- Unescape the string
- Replace ( ' ) with ( - )
- Convert non-latin characters: компьютер (Russian) -> komp'yuter (converted)
- Remove ( ' )
- ...

**Comment by un33k** [comment]
Sounds good.  I'll role it out during the next release cycle, and pull
request with fix+test is always welcomed. Thx

**Comment by fabiocaccamo** [comment]
Another idea could be to replace escaped quotes characters with ( - ): 

%27 ( ' ) -> ( - )
%22 ( " ) -> ( - )
%u2018 ( ‘ ) -> ( - )
%u2019 ( ’ ) -> ( - )
%u201C ( “ ) -> ( - )
%u201D ( ” ) -> ( - )

**Comment by fabiocaccamo** [comment]
Just made PR

**Comment by un33k** [comment]
Available in version 1.2.0. Thx.
